cmd/root.go

```diff
@@ -80,6 +84,12 @@ func (c *rootCommand) persistentPreRunE(cmd *cobra.Command, args []string) error
 		return err
 	}
 	c.globalState.Logger.Debugf("k6 version: v%s", consts.FullVersion())
+
+	err = c.setupTracerProvider()
+	if err != nil {
+		return err
+	}
+
```

Comment:
This also means that it should not be part of the globalState ... probably

Reply:
Right, I will move the initialization/stopping of the `TracerProvider` to the _run_ command.
Can you provide more details on what would be the best approach to "propagate" the tracer provider from the run command to the VU? Does it make sense to modify the `Runner` [interface](https://github.com/grafana/k6/blob/641af0a002dedc756fae6814d0b4dc739a4dd529/lib/runner.go#L54) _NewVU_, and possibly _Setup_ and _Teardown_ methods to accept a `TracerProvider` as input and propagate the `TracerProvider` through the `Scheduler` implementation? Although this might imply quite a few changes.

Otherwise, we might keep it in `GlobalState` so it's propagated to `TestPreInitState` when [loading](https://github.com/grafana/k6/blob/641af0a002dedc756fae6814d0b4dc739a4dd529/cmd/run.go#L96) the test.

Differently from the metrics, where we control the full implementation, the underlying OTEL's `TracerProvider` already supports configuring multiple exporters, so we might not need a "manager" even in the long run?

Reply:
Yeah, I looked at the code, and it will be nice if we can just use the same mechanism as the Samples channel, but it seems like a lot of work.

I am okay with you keeping at the `TestPreInitState` for this PR, but I think we should refactor this later.

This will currently work, but having this somehow get the credentials from `--out cloud` will be a nightmare.

Reply:
In relation with moving this to _run_ command and its exiting process. How should we treat traces as of now in this regard? Should we wait for all traces to be flushed before exiting, as it's done for metrics, or should we set a reasonable TO for it?

Reply:
I mean - we do wait metric outputs, so why not wait on trace outputs? 

It makes little sense in the general case to not wait to output the traces and spans we've collected. 

Maybe adding some timeout is okay, but given that we don't really have one for metric outputs I would argue this falls outside of this PR.

Just please start stopping it *concurrently* with the metric outputs so we don't wait for one and then for the other.